### PR TITLE
integrate challenge details page

### PR DESCRIPTION
**Pull request by okg00dbye**
changes:
add getUserTasksForChallenge route
add fields completed_at, forfeited_at to userChallenge
add fields creator_id, color to Challenge
integrate ChallengeMilestones to display based on BE data

### CODE REVIEW

**Comment by sebastiantoh** [comment]
I helped to merge from master as well as updated the seeds and migration files (so that creator is a FK to users rather than string). I'll be basing my PR off this branch!

File: backend/app/views/user_tasks/tasks_for_day.json.jbuilder [hunk 1]
```diff
@@ -2,8 +2,5 @@
 
 json.data @user_tasks do |user_task|
   json.id user_task.id
-  json.name user_task.task.name
-  json.challengeName user_task.task.challenge.name
-  json.description user_task.task.description
```
Comment (sebastiantoh) [inline backend/app/views/user_tasks/tasks_for_day.json.jbuilder:7]: Is this intentionally removed?

**Comment by jialin7878** [comment]
> I helped to merge from master, updated the seeds and migration files (so that creator is a FK to users rather than string) and fixed some lint errors.
> 
> I'll be basing my PR to integrate the stats tab off this branch!

tq!!

File: backend/app/views/user_tasks/tasks_for_day.json.jbuilder (hunk 1 above)
Comment (jialin7878) [reply backend/app/views/user_tasks/tasks_for_day.json.jbuilder]: Meant to remove in for challenges not for day, thanks for catch!

File: frontend/src/pages/challenge/ChallengeDetailsPage.tsx [hunk 2]
```diff
@@ -71,23 +76,34 @@ enum TabItem {
...
     dispatch(loadAllTasks(Number(challengeId)));
+    dispatch(loadUserTasksForChallenge(Number(challengeId)));
   }, []);
 
   const { challengeId } = useParams<{ challengeId: string }>();
+
+  const challenge =
+    useSelector((state: RootState) =>
+      getChallenge(state, Number(challengeId))
+    ) ?? {};
```
Comment (sebastiantoh) [inline frontend/src/pages/challenge/ChallengeDetailsPage.tsx:95]: hmm, im not sure about using {} as a default value. Would it be better to just add a 
```
if (!challenge) {
  return <Skeleton />
}
```
?

Comment (jialin7878) [reply frontend/src/pages/challenge/ChallengeDetailsPage.tsx]: updated!

**Pull request merged**